File: frontend/ksweb-app/src/app/core/auth/auth.models.ts
export interface AuthUsuario {
  usrCodigo: number;
  userId: number;
  nome: string;
  email: string;
  nivel: string;
  setor: string;
  queues: string[];
}

export interface LoginRequest {
  email: string;
  senha: string;
  manterConectado: boolean;
}

export interface AuthResponse {
  usuario: AuthUsuario;
  accessToken: string;
  expiresIn: number;
  refreshToken: string;
}

export interface MeResponse {
  usuario: AuthUsuario;
}

File: backend/KSWeb.Api/Models/AuthModels.cs
namespace KSWeb.Api.Models;

public sealed record LoginRequest(
    string Email,
    string Senha,
    bool ManterConectado);

public sealed record RefreshRequest(string RefreshToken);

public sealed record LogoutRequest(string RefreshToken);

public sealed record AuthResponse(
    AuthUsuario Usuario,
    string AccessToken,
    int ExpiresIn,
    string? RefreshToken);

public sealed record AuthUsuario(
    int UsrCodigo,
    long UserId,
    string Nome,
    string Email,
    string Nivel,
    string Setor,
    IReadOnlyList<string> Queues);

